From 94fb896c431e96db1f489e1a344e48a0a7d11209 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Wed, 10 Jan 2024 20:10:10 +0100
Subject: [PATCH 08/10] scripts/buildapi.mk: reduce the number of subprocesses

Ensure that dpkg-buildapi is only called at most once.

If so, call it once and cache the result for future expansions.
---
 scripts/mk/buildapi.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/scripts/mk/buildapi.mk b/scripts/mk/buildapi.mk
index ba6b43543..8ceb13cab 100644
--- a/scripts/mk/buildapi.mk
+++ b/scripts/mk/buildapi.mk
@@ -4,7 +4,11 @@ ifndef dpkg_buildapi.mk_included
 dpkg_buildapi.mk_included :=
 
 # Default API level when not set.
-DPKG_BUILD_API ?= $(shell dpkg-buildapi)
+# Run dpkg-buildapi if the variable is actually expanded,
+# but then replace the recursive definition with a non-recursive one
+# (and of course return the asked value).
+DPKG_BUILD_API ?= \
+  $$(eval DPKG_BUILD_API := $$(shell dpkg-buildapi))$$(DPKG_BUILD_API)
 
 # We could use only built-in GNU make functions, but that seems too much
 # complexity given no integer operators, given that we currently have to
-- 
2.39.2

